### DISCUSSION TITLE
Best Practices for Migrating Configuration from Development to Production Environments

### DISCUSSION THREAD

**Discussion by lnwara** [keycloak/keycloak#29469] [Q&A]
I have successfully set up a Keycloak instance (using the quay.io/keycloak/keycloak:23.0.0 image) in my development environment. This setup includes clients with authorization, authorization flows, and a custom theme. All components are working as expected. However, when I attempted to migrate these configurations to a pre-production environment using the partial export/import method, not all configurations were applied correctly.

Could you provide guidance on the best practices for this type of migration? Are there specific steps or tools recommended to ensure a complete and accurate configuration transfer?

**Comment by jordibuj** [comment]
So you did the export/import through the admin console? Looks like the CLI is better suited for a complete config transfer: https://www.keycloak.org/server/importExport
